lesson_2_Section_30_new.ipynb

```diff
@@ -0,0 +1,237 @@
+{
```

Comment:
<p>Good place to end the section</p><p> </p><p>Small thing, I realized Im not 100% sure what exactly is the windowed averaging that'll be shown. Do we wan't to call it that or call it nearest neighbors?  Might be nice to call it that, since it's googleable.  Could make an exercise to implement k nearest neighbor regression from scratch in python since its easy, or use scikit-learn on some data, or both and compare?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
Oh youre right I was loose with my terminology but it does open an interesting question, should we show nearest neighbors or a window?

Window might be more intuitive for introducing kernel because you just extend the window to infinity and add a weighting function, but then I guess you could say the same for N neighbors as well

And technically were not saying nearest neighbors but N to the left and right

Anyhow between the three whats your strongest preference
1. Window function
2. Nearest Neighbors
3. N closest left and right?

Mine is on 1, but only weakly so. I think 3 we should avoid entirely

Reply:
<p>I think I'm both 1 and 2.  I can see different folks coming at it from either direction.  If youre a SQL whiz, this is like window functions, or maybe you're an engineer and you're thinking about signal processing or spectral stuff.  If you've taken machine learning 101 somewhere, you are maybe thinking "is this not KNN?"  If neither, 'nearest neighbor' at least has a more literal interpretation, where 'window function' might sound a bit mysterious.  I think at least in a video however it's explained it should be mentioned at least as a parenthetical "by the way yes, this is KNN regression".</p>

Reply:
great refinement, Ill get that added into the script in the right place

Reply:
> Could make an exercise to implement k nearest neighbor regression from scratch in python since its easy, or use scikit-learn on some data, or both and compare? 

Good idea, I do it a very hacky but we can get the students do it using vectorization